### PR TITLE
FHAC-201:Universal Client DocumentRetrieve Service

### PR DESCRIPTION
**Pull request by alameluchidambaram**
1. Created DR service for Universal Client in AdminGUI.
2. DR specific Request, Response are built within DocumentRetrieveCore.

### CODE REVIEW

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/messaging/builder/impl/dr/DocumentRetrieveRequestBuilderImpl.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.docretrieve.messaging.builder.impl.dr;
```
Comment (christophermay07) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/messaging/builder/impl/dr/DocumentRetrieveRequestBuilderImpl.java]: Why does this package name end with ".dr" when "docretrieve" is already part of the package name?

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/messaging/builder/impl/dr/DocumentRetrieveRequestBuilderImpl.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import ihe.iti.xds_b._2007.RetrieveDocumentSetRequestType;
+import ihe.iti.xds_b._2007.RetrieveDocumentSetRequestType.DocumentRequest;
+
+/**
+ *
+ * @author achidamb
+ */
+public class DocumentRetrieveRequestBuilderImpl implements DocumentRetrieveRequestBuilder {
+
+    private String HCID;
```
Comment (christophermay07) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/messaging/builder/impl/dr/DocumentRetrieveRequestBuilderImpl.java]: We typically use all-lower "hcid" as a variable consistently throughout CONNECT (with all-caps naming for final vars)

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/messaging/builder/DocumentRetrieveRequestBuilder.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    /**
+     *
+     * @return The DocumentRetrieve Request will be returned
+     */
+    public RetrieveDocumentSetRequestType getMessage();
+
+    /**
+     *
+     * @param HCID sets the HCID for the DocumentRetrieve Request
```
Comment (christophermay07) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/messaging/builder/DocumentRetrieveRequestBuilder.java]: "param" names and describes a parameter -- the description of the function as a whole should be above on its own line(s).  Please fix this across all files.

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/builder/impl/dr/DocumentRetrieveResultsModelBuilderImpl.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import gov.hhs.fha.nhinc.docretrieve.model.DocumentRetrieveResults;
+import gov.hhs.fha.nhinc.docretrieve.model.builder.DocumentRetrieveResultsModelBuilder;
+import ihe.iti.xds_b._2007.RetrieveDocumentSetResponseType;
+import ihe.iti.xds_b._2007.RetrieveDocumentSetResponseType.DocumentResponse;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.activation.DataHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
```
Comment (christophermay07) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/builder/impl/dr/DocumentRetrieveResultsModelBuilderImpl.java]: We're not using SLF4J yet, please switch to log4j

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/DocumentRetrieveResults.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
```
Comment (christophermay07) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/DocumentRetrieveResults.java:26]: Wrong license

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/builder/impl/dr/DocumentRetrieveResultsModelBuilderImpl.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+/**
+ *
+ * @author achidamb
+ */
+public class DocumentRetrieveResultsModelBuilderImpl implements DocumentRetrieveResultsModelBuilder {
+
+    private static final Logger LOG = (Logger) LoggerFactory.getLogger(DocumentRetrieveResultsModelBuilderImpl.class);
```
Comment (christophermay07) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/builder/impl/dr/DocumentRetrieveResultsModelBuilderImpl.java]: Remove the unneeded cast

File: Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/builder/impl/dr/DocumentRetrieveResultsModelBuilderImpl.java (hunk 4 above)
Comment (jsmith2) [inline Product/Production/Services/DocumentRetrieveCore/src/main/java/gov/hhs/fha/nhinc/docretrieve/model/builder/impl/dr/DocumentRetrieveResultsModelBuilderImpl.java]: Good with pr with fixes suggested by Chris/Naresh.

**Comment by TabassumJafri** [comment]
Agrees with team comments. Apart from that PR looks good

**Comment by nareshcgi** [comment]
I am good with the PR.

**Pull request merged**